licenses/library.go

```diff
@@ -117,7 +118,33 @@ func Libraries(ctx context.Context, classifier Classifier, includeTests bool, ig
 			klog.Errorf("Package %s does not have module info. Non go modules projects are no longer supported. For feedback, refer to https://github.com/google/go-licenses/issues/128.", p.PkgPath)
 			return false
 		}
-		licensePath, err := Find(pkgDir, p.Module.Dir, classifier)
+
+		stopAt := p.Module.Dir
+		if isVendorModule(newModule(p.Module), pkgDir) {
+			splits := strings.SplitN(pkgDir, "/vendor/", 2)
+			moduleSplits := strings.Split(p.Module.Path, "/")
+			// There seems to be 3 common cases in terms of go module directory structure:
```

Comment:
I am not overly happy/attached to this code, but tried to document what I saw when running these build thru our build pipelines.  I think the main fix here is by stopping the traversal up at the vendor directory.  We could remove all this and just use that instead of trying to be "smart" by making these assumptions.  I would not guarantee these are assumptions are even correct.  There could def be other cases where the folder structure is deeper or shallower for legit reason.

Looking for feedback here for sure.